### PR TITLE
Rhythm fields and color-switch floor refactor

### PR DESCRIPTION
**Pull request by silicons**
Color floors refactored a bit.
Added rhythm fields, which force everything inside them to move/act at a certain speed governed by their tick rate.
Now supports beatmap import to sync with sounds.

### CODE REVIEW

File: code/__DEFINES/subsystems.dm [hunk 1]
```diff
@@ -85,13 +85,14 @@
 #define FIRE_PRIORITY_THROWING		6
 #define FIRE_PRIORITY_FLIGHTPACKS	7
 #define FIRE_PRIORITY_SPACEDRIFT	7
-#define FIRE_PRIOTITY_SMOOTHING		8
```
Comment (praisenarsie) [inline code/__DEFINES/subsystems.dm]: titty

**Comment by cebutris** [comment]
King Konga mode where every eighth "beat" is skipped when

**Comment by Cyberboss** [comment]
Stale

File: code/controllers/subsystem/npcpool.dm [hunk 2]
```diff
@@ -135,14 +130,23 @@ SUBSYSTEM_DEF(npcpool)
...
 		if(prob(helpProb) && candidate.takeDelegate(check,FALSE))
 			--canBeUsed.len
 			candidate.eye_color = "yellow"
 			candidate.update_icons()
-			
+
 		if(!currentrun.len || MC_TICK_CHECK)	//don't change SS state if it isn't necessary
 			return
 
+/datum/controller/subsystem/npcpool/proc/process_simple_animal(mob/living/simple_animal/SA)
```
Comment (vuonojenmustaturska) [inline code/controllers/subsystem/npcpool.dm]: no

**Comment by optimumtact** [comment]
This is way way way to invasive for the feature set it brings.

**Pull request closed without merging**